File: 744.寻找比目标字母大的最小字母.java
/*
 * @lc app=leetcode.cn id=744 lang=java
 *
 * [744] 寻找比目标字母大的最小字母
 */

// @lc code=start
class Solution {
    public char nextGreatestLetter(char[] letters, char target) {
        if (target >= letters[letters.length - 1]) {
            return letters[0];
        }
        int low = 0, high = letters.length - 1;
        int mid = 0;
        while (low < high) {
            mid = (low + high) / 2;
            if (letters[mid] <= target) {
                low = mid + 1;
            } else {
                high = mid;
            }
        }
        return letters[high];

    }
}
// @lc code=end

File: 744.寻找比目标字母大的最小字母.cpp
/*
 * @lc app=leetcode.cn id=744 lang=cpp
 *
 * [744] 寻找比目标字母大的最小字母
 */

// @lc code=start
class Solution
{
public:
    char nextGreatestLetter(vector<char> &letters, char target)
    {
        int n = letters.size();
        int l = 0, h = n - 1;
        while (l <= h)
        {
            int m = l + (h - l) / 2;
            if (letters[m] <= target)
            {
                l = m + 1;
            }
            else
            {
                h = m - 1;
            }
        }
        return l < n ? letters[l] : letters[0];
    }
};
// @lc code=end
